File: android/DetonatorGallery/src/main/java/com/iconshot/detonator/gallery/GalleryAlbumReaderCloseRequest.java
package com.iconshot.detonator.gallery;

import com.iconshot.detonator.Detonator;
import com.iconshot.detonator.request.Request;

public class GalleryAlbumReaderCloseRequest extends Request<GalleryAlbumReaderCloseRequest.CloseData> {
    public GalleryAlbumReaderCloseRequest(Detonator detonator, IncomingRequest incomingRequest) {
        super(detonator, incomingRequest);
    }

    @Override
    public void run() {
        CloseData data = decodeData(CloseData.class);

        GalleryAlbumReaderReadRequest.albumListMap.remove(data.id);

        end();
    }

    protected static class CloseData {
        int id;
    }
}

File: ios/DetonatorGallery/Sources/DetonatorGallery/GalleryAlbumReaderCloseRequest.swift
import Photos

import Detonator

public class GalleryAlbumReaderCloseRequest: Request {
    public override func run() -> Void {
        let data: CloseData = decodeData()!
                
        GalleryAlbumReaderReadRequest.fetchResults[data.id] = nil
        
        end()
    }
    
    public struct CloseData: Decodable {
        let id: Int
    }
}
